Fix CI Failures

philipc2 opened the pull request:
<!--  The PR title should summarize the changes, for example "Add `Grid._build_face_dimension` function".
      Avoid non-descriptive titles such as "Addresses issue #229". -->

<!--  Replace XXX with the number of the issue that this PR will resolve. If this PR closed more than one,
      you may add a comma separated sequence-->
Closes #723

## Overview
<!--  Please provide a few bullet points summarizing the changes in this PR. This should include
      points on any bug fixes, new functions, or other changes that have been made. -->
* Modifications to Point Raster functionality to conform to new Dask changes. 
* Removes some old pins that weren't in `enviroment.yml`

erogluorhan reviewed (commented):
An actual solution, if possible, rather than pinning would be ideal. Let me know if you explored anything that might be related to the actual issue; I am happy to look into this a bit.

philipc2 commented:
> An actual solution, if possible, rather than pinning would be ideal. Let me know if you explored anything that might be related to the actual issue; I am happy to look into this a bit.

I can continue to do some digging into it, the failure is as follows:

> ../../../miniconda3/envs/uxarray_build/lib/python3.10/site-packages/datashader/data_libraries/dask.py:147: AssertionError

It appears to be something with Datashader and Dask. I'll continue looking into it and try to reproduce it locally.

philipc2 commented:
This seems to be because of the user of `dask-expr` in the recent release for Dataframes, which breaks something with the way that we are caching (and honestly, we should investigate better ways to "cache" as opposed to just keeping a reference in the grid)
https://docs.dask.org/en/stable/changelog.html#v2024-3-0

Considering how quick point rasterization is, I don't think we need to cache, so I've removed it and cleaned up the point raster. 

![image](https://github.com/UXARRAY/uxarray/assets/67855069/2ba2101e-8bc3-4801-a18e-c4032010c752)

![image](https://github.com/UXARRAY/uxarray/assets/67855069/2902edb6-3415-4976-a9e9-e5570a0fcffd)

Any thoughts?

erogluorhan commented on uxarray/grid/grid.py:
```diff
@@ -143,8 +143,6 @@ def __init__(
         self._poly_collection = None
         self._line_collection = None
-        self._centroid_points_df_proj = [None, None]
-        self._corner_points_df_proj = [None, None]
```
These were not used at all, right?

erogluorhan commented on uxarray/plot/dataarray_plot.py:
```diff
@@ -214,65 +213,29 @@ def _point_raster(
-
-        # Construct Dask DataFrame
-        point_ddf = dd.from_dict(data=point_dict, npartitions=npartitions)
+    if projection is not None:
```
This change seems to be beyond the scope; please clarify

philipc2 replied:
There's no changes to the functionality, other than this PR does not cache the results due to the issues with Dask.

#739 will address this in the future, after some research into best ways to handle storing these dask objects.

erogluorhan replied:
I am so sorry; I missed your [yesterday's comment](https://github.com/UXARRAY/uxarray/pull/731#issuecomment-2013472875), which clarifies everything, before giving this review.

philipc2 replied:
No worries!

erogluorhan commented on uxarray/plot/dataarray_plot.py:
```diff
@@ -214,65 +213,29 @@ def _point_raster(
 
-        hv.extension("bokeh")
-        points = hv.Points(point_ddf, ["lon", "lat"]).opts(size=size)
+    # this will be fixed in #733
```
Please clarify

philipc2 replied:
#733 will prevent the need to always call `hv.extension`, instead it will only be called whenever we swap from one backend to another.

erogluorhan commented on uxarray/plot/dataarray_plot.py:
```diff
@@ -214,65 +213,29 @@ def _point_raster(
-            else:
-                uxda.uxgrid._corner_points_df_proj[0] = point_ddf
-                uxda.uxgrid._corner_points_df_proj[1] = projection
+    point_dict = {"lon": lon, "lat": lat, "var": uxda.data}
```
Please clarify

philipc2 replied:
As per other comments, this removes the caching, since our approach using the Dask dataframe is leading to issues.

erogluorhan reviewed (commented):
Thanks for this! However, changes in this PR seem to be beyond this PR's scope, or it is hard to understand all those changes were necessary for the Dask issue and if they are not changing the code flow significantly. Please clarify

philipc2 commented:
> Thanks for this! However, changes in this PR seem to be beyond this PR's scope, or it is hard to understand all those changes were necessary for the Dask issue and if they are not changing the code flow significantly. Please clarify

Thanks for the review! I'll add some comments describing the sections of the code. However, these aren't new. The changes made here are, for the time being, remove our "caching" approach for the point rasters. When we use a cached dataframe, attempting to swap the data variable for the points leads to the Dask error that we were getting.

The difference in performance with or without using caching is comparable as seen in the Viz cookbook,.

Additionally, there are probably better ways for caching compared to the way we were doing it, which worked fine for standard arrays, but not so much for Dask.

Let me know what you think, I submitted an issue (#739) that will describe this in detail.

erogluorhan reviewed (approved):
So sorry for missing your [yesterday's comment](https://github.com/UXARRAY/uxarray/pull/731#issuecomment-2013472875) and asking clarifications. Everything is clarified now.
